src/main/java/graphql/execution/incremental/IncrementalCallState.java

```diff
@@ -9,21 +9,32 @@
 
 import java.util.Collection;
 import java.util.Collections;
+import java.util.Comparator;
 import java.util.Deque;
+import java.util.Queue;
 import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.PriorityBlockingQueue;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicInteger;
 
 import static graphql.incremental.DelayedIncrementalPartialResultImpl.newIncrementalExecutionResult;
 
 /**
- * This provides support for @defer directives on fields that mean that results will be sent AFTER
- * the main result is sent via a Publisher stream.
+ * Manages the state of incremental calls (for fields that have been @defer'red or @stream'ed), which will result
+ * in data that is sent back to the client AFTER the initial response,
+ * via a {@link Publisher} of {@link DelayedIncrementalPartialResult}.
  */
 @Internal
 public class IncrementalCallState {
     private final AtomicBoolean incrementalCallsDetected = new AtomicBoolean(false);
-    private final Deque<IncrementalCall<? extends IncrementalPayload>> incrementalCalls = new ConcurrentLinkedDeque<>();
+
+    // PriorityBlockingQueue doesn't have a constructor that accepts just the comparator and
+    // uses the default initial capacity. 11 is the default value used in the implementation I'm looking at,
+    // and it seems reasonable for our use case.
+    private static final int QUEUE_INITIAL_CAPACITY = 11;
+    private final Queue<IncrementalCall<? extends IncrementalPayload>> incrementalCalls =
+            new PriorityBlockingQueue<>(QUEUE_INITIAL_CAPACITY, Comparator.comparingInt(IncrementalCall::level));
```

Comment:
Priority queue ordered by field level to ensure fields in higher levels don't resolve before fields in lower levels. This is a possibility when we have nested defers, and could be reproduced in a test.